python/tvm/relay/op/contrib/tidl.py

```diff
@@ -129,155 +171,79 @@ def _dense_relu_pattern():
         relu_out = is_op('nn.relu')(dense_out)
         return relu_out
 
+    def _dense_relu_checker(extract):
+        dense_op = extract.args[0]
+        return _dense_whitelist_fn(dense_op.attrs, dense_op.args)
+
     #relu has to be preceded by dense and bias_add
     def _dense_bias_relu_pattern():
         dense_out = is_op('nn.dense')(wildcard(), is_constant())
         bias_out = is_op('nn.bias_add')(dense_out, is_constant())
         relu_out = is_op('nn.relu')(bias_out)
         return relu_out
 
+    def _dense_bias_relu_checker(extract):
+        dense_op = extract.args[0].args[0]
+        return _dense_whitelist_fn(dense_op.attrs, dense_op.args)
+
     #relu has to be preceded by dense and bias_add
     def _dense_add_relu_pattern():
         dense_out = is_op('nn.dense')(wildcard(), is_constant())
         add_out = is_op('add')(dense_out, is_constant())
         relu_out = is_op('nn.relu')(add_out)
         return relu_out
 
+    def _dense_add_relu_checker(extract):
+        dense_op = extract.args[0].args[0]
+        return _dense_whitelist_fn(dense_op.attrs, dense_op.args)
+
     #bias_add has to be preceded by dense
     def _dense_bias_pattern():
         dense_out = is_op('nn.dense')(wildcard(), is_constant())
         bias_out = is_op('nn.bias_add')(dense_out, is_constant())
         return bias_out
 
+    def _dense_bias_checker(extract):
+        dense_op = extract.args[0]
+        return _dense_whitelist_fn(dense_op.attrs, dense_op.args)
+
     #bias_add has to be preceded by dense
     def _dense_add_pattern():
         dense_out = is_op('nn.dense')(wildcard(), is_constant())
         add_out = is_op('add')(dense_out, is_constant())
         return add_out
 
+    def _dense_add_checker(extract):
+        dense_op = extract.args[0]
```

Comment:
Same as above. We nee to have the following:
if not any([isinstance(arg, tvm.relay.expr.Constant) for arg in add.args]):
    # must be bias_add here
    return False